# Contents


# Variant calling phenogram
fish_phenogram.py # Plotting sperm pools data from variant calling pipeline

# Go terms plots
autobubble_goplot.py # Uses a ShinyGo csv file as input to automate and control bubble plots in python

# Retroseq analysis

testing_glm.R # For GLM analysis of TE counts that have been filtered for read count and high confidence, and have had the control group TEs removed in the experimental groups

For Retroseq derived input vcf files:
phenogram.py # Phenogram plotting with chromatin accessibility, gene labels, centromere locations etc.

Files needed for phenogram.py:

Your.vcf

chrcen.txt # DanRer11 compatible 

genes.txt # DanRer11

chrom_end.txt # Danrer11

filter_retroseq.py filtering

class_chart.py 

TE_Chrom_count.py

# Snpeff and SnpSift
vcfinput_phenogram.py # Phenogram

filter.py # Filter vcf file output by impact/terms e.g to filter High impact variants



